athletic_pandas/models.py

```diff
@@ -1,22 +1,36 @@
 from . import algorithms
 from .base import BaseWorkoutDataFrame
 from .helpers import requires
+from vmpy.metrics import normalized_power, power_duration_curve
+from vmpy.streams import compute_zones, wpk
 
 
 class WorkoutDataFrame(BaseWorkoutDataFrame):
     _metadata = ['athlete']
 
+    @requires(columns=['power'])
```

Comment:
In general, I think the require decorator limits the usage. If you get data from Strava for example, the power stream is called 'watts' there. I would propose to ditch require in favor of sane defaults with an ability to override during the function call. Somethin along the line wdf.calculate_awesome_power(arg='power', **kwargs)

In this particular case I've removed require for an athlete to allow optionally specifying 'ftp' using keywords. It falls back on athlete['ftp'] as s default.

Reply:
I think this is really similar to the discussion about accepting multiple input argument types and I am not in favor of it. I think as a developer you have a responsibility to make the interface of you library as well defined as possible, accepting multiple input types or accepting that certain data is stored in different columns makes the interface unclear and therefore more difficult to use. For me this would also make the purpose of the dataframe redundant as you cannot just use the class methods "as is" without understanding the underlying data structure.
Anyway, I am writing a longer response where I will dive into this (even) more...

Reply:
> accepting that certain data is stored in different columns makes the interface unclear and therefore more difficult to use

The problem with too simple interfaces is that they limit the use. In case of requiring 'power' only you force the user in figuring out how to rename the columns in pd.DataFrame to get it working. The *power/watts* duality is a one example forced on us by Strava, but with *heartrate* you will get a full multitude of the possibilities: {hr, bpm, heart_rate, you_name_it}

The methods can simply be written with defaults in mind as `def method(self, arg='power')`, and can be called as `wdf.method()` with defaults and with a user-defined name for other cases `wdf.method(arg='watts')`. This pattern is in fact not that different from how pandas already exposes matplotlib methods. 

Another aspect is a sole reliance on the Athlete obj. I think it is less canonical and therefore less obvious for the user to expect that only way to parametrize wdf methods is through the Athlete object. It is a great placeholder for defaults, but it is also quite natural to expect the parameters to be exposed as a function signature. 

Imagine you want to compare two Wbal calculations on the same data set, but with different W' and CP values. The only way to do it now is through redefining the Athlete properties, while I would naturally expect that W' and CP' should be available as inputs to the method call.